### ISSUE TITLE
[Request]: QoL - Collapsible Pre-download box

### ISSUE THREAD

**Issue by Kiki79250CoC** [CollapseLauncher/Collapse#537]
### Is your request related to a problem?

No

### How is the feature request related to the problem?

This is a (I think) relatively simple QoL change that can help to have a cleaner experience while using Collapse.

### Describe your proposed solution

The concept is to simply "minimize" the "update available box in the launcher.

**Figure 1:** How the Update box is as of today (1.80.15/16d)
![Collapse Preview 1_b1](https://github.com/user-attachments/assets/9adcc717-b555-4e02-a9a4-7e99ba693dfb)

The idea is just to, instead of completely wipe that box chen click on "close", to "move" it to the far-right part, at the level of the social media panel with a simple button indicating that it is still available.

**Figure 2:** Illustrating what the concept technically should be seen for "popup minimization"
![Collapse Preview 1_b2](https://github.com/user-attachments/assets/01c8f010-8176-4aad-9103-cf856f34f7b6)
**Figure 3:** Mockup (made in GIMP) of how it would probably look like
![Collapse Preview 2_b1](https://github.com/user-attachments/assets/e716de93-b511-4ab0-9f24-9b5af5cfe15f)
Eventually, this item is a button that can be clicked to "restore" the popup and perform actions such as verify or simply the pre-download.

**Figure 4:** Illustrating what the concept technically should be seen for "popup restauration"
![Collapse Preview 2_b2](https://github.com/user-attachments/assets/ef47c097-1b82-4584-9a10-3a4a23ecb5ec)
**Figure 5:** Basically ***Figure 1***, because it's the "original" state.

--

This method ultimately have some advantages:
- It uses less space on-screen
- It allows to show that popup again without having to switch region/restart Collapse
- Allows the "Pre-download panel" to be "hidden" while performing actions while still having something indicating it's presence.
  - In that case, a warning may be added if the user tries to close Collapse while the download is still in progress.
  - That warning may also automatically restore the popup if "minimized".
- No idea yet, but maybe the icon can be animated while an action is performed like the arrow moving during download.

--

For small window option, that thing can still be implemented, but it would require to decrease the size of the social media panel to make it fit.

**Figure 6:** Same Mockup as ***Figure 3*** but in small window mode
![Mini](https://github.com/user-attachments/assets/673bd1bc-b853-4a23-bc25-283824a702b7)
Eventually, some things are to take on count, such as if the "hidden" state should be persistent or not, or if it only persist for "next version" then revert to default (visible) when next version pre-download is available (for example if you hidden it for Genshin 4.8 pre-download, it reverts to the default setting when v5.0 pre-download is available*).

For convenience if the setting is persistent, it should be per-game as someone would probably want show full banner on HSR but not in Genshin.

\* I'm probably not clear, sorry, I just can't explain things with simplicity.

### Describe alternatives you've considered

An alternative would just be as simple as a "collapse" button to shrink the box to leave only the bare minimum in it

Basically the box would consists of just a line of text with:
> Downloading Preload Package (Sophon Mode - 30/11696) - 1% (57.4 MB/12.25 GB - 40.37 MB/s) - ETA 17m40s

This idea is less Aesthetic but if the goal is just compacting things, it'll do the job.

### Additional context

n/a

**Comment by bagusnl** [comment]
I think instead of making new badge in the page, we can use the existing background download function and put the preload download in the notification modal when user minimized it. wyt?

Cc @neon-nyan

**Comment by neon-nyan** [comment]
@bagusnl 
The design idea should work for this one. We will be planning it in the future.
